### PR TITLE
Add a test.

### PR DESCRIPTION
**Pull request by bangerth**
This verifies a question Timo just asked about multiplying a tensor by an integer.

### CODE REVIEW

File: tests/base/product_type_04.cc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    Tensor<1,3> t;
+    t[0] = 1;
+    t[1] = 2;
+    t[2] = 3;
+    Assert (7*t == 7.0 * t, ExcInternalError());
+    Assert (t*7 == t * 7.0, ExcInternalError());
+    Assert (t*7 == 7*t    , ExcInternalError());
+    Assert (t*7 == t+t+t+t+t+t+t, ExcInternalError());
+  }
+
```
Comment (Rombur) [inline tests/base/product_type_04.cc]: Don't you want to use AssertThrow ? This test does nothing in release mode, doesn't it ?

**Comment by tjhei** [comment]
You are using whole numbers for everything in this test. Wouldn't it make sense to use doubles for the Tensor entries to see how rounding is happening?

**Comment by bangerth** [comment]
Both good points. I'm now running the test only in debug mode, and I'm using non-integral numbers. (The latter required modifying one of the checks for round-off.)

I'm pushing a second commit that checks the same thing for symmetric tensors and a third that renames some of the tests to use underscore rather than hyphen.

**Comment by tjhei** [comment]
Great, thanks!

**Pull request merged**